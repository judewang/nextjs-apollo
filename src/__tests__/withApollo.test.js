import 'isomorphic-unfetch';
import _ from 'lodash';
import http from 'http';
import ws from 'ws';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import express from 'express';
import PropTypes from 'prop-types';
import { ApolloServer, PubSub } from 'apollo-server-express';
import { withApollo, initApollo, JsonWebToken } from '..';

const pubsub = new PubSub();

const toValues = jest.fn(_.identity);
const toModel = jest.fn(_.identity);
const fetchCorrelation = jest.fn(values => Promise.resolve(values));
const createCorrelation = jest.fn(values => Promise.resolve(values));
const updateCorrelation = jest.fn(values => Promise.resolve(values));
const resolve = jest.fn(() => 1);

const jwt = new JsonWebToken({
  JWT_SECRET: 'XYZ',
}, {
  toValues,
  toModel,
  createCorrelation,
  updateCorrelation,
  fetchCorrelation,
});

const typeDefs = gql`
  type Query { viewer: Int }
  type Subscription { viewer: Int }
`;

const resolvers = {
  Query: {
    viewer: (payload, args, context) => resolve(context),
  },
  Subscription: {
    viewer: {
      subscribe: (payload, args, context) => {
        pubsub.publish('NEW_SUBSCRIPTION', context);
        return pubsub.asyncIterator(['NEW_VIEWER']);
      },
    },
  },
};

const apolloServer = new ApolloServer({
  typeDefs,
  resolvers,
  context: jwt.contextParser(),
});

const app = express();
apolloServer.applyMiddleware({ app });

const server = http.createServer(app);
apolloServer.installSubscriptionHandlers(server);
server.listen();

const { port } = server.address();

describe('withApollo', () => {
});
