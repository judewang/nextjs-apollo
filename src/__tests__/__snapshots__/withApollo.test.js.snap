// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withData process.browser is false when NoSSR 1`] = `
Object {
  "apolloState": Object {},
  "value": 99,
}
`;
